manager/dispatcher/dispatcher.go

```diff
@@ -313,11 +323,19 @@ func (d *Dispatcher) Stop() error {
 		return errors.New("dispatcher is already stopped")
 	}
 
-	log := log.G(d.ctx).WithField("method", "(*Dispatcher).Stop")
-	log.Info("dispatcher stopping")
+	// Set shutdown to true.
+	// This will prevent RPCs that start after stop() is called
+	// from making progress and essentially puts the dispatcher in drain.
+	d.shutdown = true
```

Comment:
don't see any place where the shutdown is set back to false. Remember that the Dispatcher is not recreated but is reused.

Reply:
good point! Will address this.

Reply:
I'm a little worried by the fact that the CI was green...

Reply:
Yea, unfortunately the unit-test always creates a new dispatcher. I know.